### DISCUSSION TITLE
Cache issues?

### DISCUSSION THREAD

**Discussion by NiklasBr** [gotenberg/gotenberg#560] [Q&A]
I'm rather new to the Gotenberg ecosystem, so I am probably doing something wrong. But here goes. Any time I make a change to a `footerTemplate` och `headerTemplate` and then run a request to `/forms/chromium/convert/url` I have to wait many minutes before Chromium pick up the changes. Inspecting the body of the content makes it look like the changes are actually sent to Gotenberg, so I think that maybe there is some cache on the Goteberg/chromium side which fails to pick up these changes. Is there some setting I can try to tune to see what happens?

**Comment by gulien** [comment]
Hello @NiklasBr,

There is no cache on Gotenberg, maybe there is some on Chromium, but, I never had such issue with the header/footer feature.

You could set the logging to debug and see if the templates are the old/new ones in the logs. 🤔

**Reply by NiklasBr** [reply]
I'm a bit unsure how to do that with the PHP library. But I have a breakpoint at:

```php
$renderedHtml = Document\Service::render($snippet, [], false, [], $params);
$request = Gotenberg::chromium($apiUrl)->header(Stream::string('header.html', $renderedHtml));
```

At which point I can see that the parameter `$renderedHtml` is updated as expected.

**Reply by gulien** [reply]
Actually you won’t be able to do that with the PHP library 😊 when launching the Gotenberg container, you may provide a flag to set the logging to debug. See https://gotenberg.dev/docs/modules/logging
